bittensor/commands/root.py

```diff
@@ -520,12 +528,21 @@ def _run(cli: "bittensor.cli", subtensor: "bittensor.subtensor"):
             cli.config.weights = Prompt.ask(f"Enter weights (e.g. {example})")
 
         # Parse from string
-        netuids = np.array(
-            list(map(int, re.split(r"[ ,]+", cli.config.netuids))), dtype=np.int64
+        matched_netuids = list(map(int, re.split(r"[ ,]+", cli.config.netuids)))
+        netuids = (
+            torch.tensor(matched_netuids, dtype=torch.long)
+            if os.environ.get("USE_TORCH")
+            else np.array(matched_netuids, dtype=np.int64)
         )
-        weights = np.array(
-            list(map(float, re.split(r"[ ,]+", cli.config.weights))),
-            dtype=np.float32,
+
+        matched_weights = list(map(float, re.split(r"[ ,]+", cli.config.weights)))
```

Comment:
just a nitpick: I would prefer to use a list comprehension for its readability and style, which aligns with Python's recommendations (PEP 8). In terms of performance, both options are practically the same.
`matched_weights = [float(weight) for weight in re.split(r"[ ,]+", cli.config.weights)]`